### DISCUSSION TITLE
various questions regarding the tab component

### DISCUSSION THREAD

**Discussion by ysk3a** [NG-ZORRO/ng-zorro-antd#8812] [Q&A] [answered]
hi,
i had a few questions about ng.ant.design tabs i was wondering about.

1). tab item styling:
is it possible to left justify the text of the tab instead of the default centered?
i could manually change the style such that it looks left justified using either ngdeep or nzTabBarStyle like in my demo
https://stackblitz.com/edit/angular-1qq3tj-batt61
but i am wondering if there was an api input that already does this that would decrease the use of ng-deep.

***

2). customizing tab width:
the documentation examples seems to show that the `nz-tab` width will grow based on the content.
however, there does not seem to be any api talking about tab width or max-width or maybe i am blind?
unless the only way to do this, is to manually set it using classes and ng-deep styles?

***

3). customizing menu when there are lots of tabs:
is it possible to customize the menu content?
for example, in a tabset with overflowing tabs like https://ng.ant.design/components/tabs/en#components-tabs-demo-slide ,
i would like to,
a) show ALL menu items instead of the default behavior i.e. shows menu items not in the viewable area.
b) add a search input box at the top of the menu popup when it opens to allow quick search/filter menu items.

***

4). tab paginating:
is there a paginator button option that the developer can set? 
this request/issue is not as big of a problem but i just wanted to know if there was such option.
by default in the example, it is not clear to the end user that there is tabs that is overflowed and requires user to scroll. 
it would be nice to have something like firefox's pagination.
![image](https://github.com/user-attachments/assets/b56ca377-9e5d-4cee-a3dc-cfce6897b7fb)
of course an alternative approach, which i think may look cleaner and would be related to 3) , would be like chromium browser where there is a dedicated dropdown menu button at the far ends of the browser that shows all tabs and a searchbar.
![image](https://github.com/user-attachments/assets/5ae91819-a0a6-419c-b1a3-5d625a9d2ba0)

the later seems the closest as ngant tab already has nzTabBarExtraContent, but it seems like `nzTabBarExtraContent` only adds extra content on the right side instead of being able to add to the left side as well?

**Comment by HyperLife1119** [comment, marked as answer]
Added support for left extraContent in #9097 :)
